### ISSUE TITLE
Removing ddclient and port forwarding all http/https

### ISSUE THREAD

**Issue by dtroose** [saltyorg/docs#317]
On [Installing Saltbox on a home server](https://docs.saltbox.dev/reference/guides/chazguides/home-server/) I have found the following issues;

First, the instructions indicate removing `ddclient` role, but there are no commands indicating how this should be done. After looking through all of the `sb` commands, including seeing it on `sb list` I gave up and just deleted the `ddclient` folder. This didn't work as apparently `sb install ddns` has a dependency (or at least a reference to) `ddclient`. Please clarify (or create a separate FAQ page) on how to remove 'roles'.

Second, it seems like it's worth clarification of what seems to be suggested under the Port Forwarding section. This seems to be saying that all http:// and https:// communication coming into the network be sent directly to the saltbox. This seems like a huge requirement. When I attempted to set the rule my router warned me with the following: "Warning: The change you are attempting to make may cause AT&T U-verse TV to stop working properly." I don't use this service, but it appears to me that it could interrupt a lot more communications than that. I have lots of clients in my house that could use http/s protocols, so it seems weird to forward all of them to just this one device. If this is as big of a deal as it seems, it's probably worth mentioning at the top of this page as well as early on in the main 'Install' page for local users.

Finally, my install seems to be running just fine, no failed packages, and yet there is no troubleshooting offered to confirm how to test the server from even the same machine as the running packages.

**Comment by saltydk** [comment]
ddns has nothing to do with ddclient, you should ask for help on our discord as support is generally not done on Github.

**Comment by saltydk** [comment]
The whole section here
> When I attempted to set the rule my router warned me with the following: "Warning: The change you are attempting to make may cause AT&T U-verse TV to stop working properly." I don't use this service, but it appears to me that it could interrupt a lot more communications than that. I have lots of clients in my house that could use http/s protocols, so it seems weird to forward all of them to just this one device.

Seems to suggest that you have little idea what a port forward is as it will not affect a single client on the same network.

**Comment by saltydk** [comment]
@chazlarson You can decide if this is a docs issue or not but I would close it.

**Comment by dtroose** [comment]
> ddns has nothing to do with ddclient, you should ask for help on our discord as support is generally not done on Github.

Per the referenced page:

> If you use Cloudflare for DNS, use the ddns saltbox role and it will be configured for you. if the ddclient role has been installed, remove it first.

Maybe that last phrase should just be removed, then?

**Comment by saltydk** [comment]
I was mostly responding to your dependency bit. They both do a similar thing which is why it asks you to ensure you're only running one of them, both are manually installed and aren't part of what the core tags install.

**Comment by dtroose** [comment]
> The whole section here
> 
> > When I attempted to set the rule my router warned me with the following: "Warning: The change you are attempting to make may cause AT&T U-verse TV to stop working properly." I don't use this service, but it appears to me that it could interrupt a lot more communications than that. I have lots of clients in my house that could use http/s protocols, so it seems weird to forward all of them to just this one device.
> 
> Seems to suggest that you have little idea what a port forward is as it will not affect a single client on the same network.

You may be right, and I might be screwing up the terminology. What I'm talking about is external https requests coming in over external port 443 would be forced to the saltbox. If I have a different https request for a different server on my local network wouldn't that isolate that other server?

**Comment by saltydk** [comment]
The bit about how to remove apps is valid though, I think we don't cover that in the docs @chazlarson

**Comment by saltydk** [comment]
> > The whole section here
> > > When I attempted to set the rule my router warned me with the following: "Warning: The change you are attempting to make may cause AT&T U-verse TV to stop working properly." I don't use this service, but it appears to me that it could interrupt a lot more communications than that. I have lots of clients in my house that could use http/s protocols, so it seems weird to forward all of them to just this one device.
> > 
> > 
> > Seems to suggest that you have little idea what a port forward is as it will not affect a single client on the same network.
> 
> You may be right, and I might be screwing up the terminology. What I'm talking about is external https requests coming in over external port 443 would be forced to the saltbox. If I have a different https request for a different server on my local network wouldn't that isolate that other server?

This is a you problem as that is outside of the scope of our project. If you want to do advanced network routing then you would need to know how to handle the proxying required.

**Comment by dtroose** [comment]
> I was mostly responding to your dependency bit. They both do a similar thing which is why it asks you to ensure you're only running one of them, both are manually installed and aren't part of what the core tags install.

Good to know! But how do you "remove" one, as currently stated on the page? Is there a "sb <command>" i'm missing?

**Comment by saltydk** [comment]
In the case of ddns and ddclient they are both docker containers, so you would just remove them with the generic 
```
docker rm -f containername
```

**Comment by dtroose** [comment]
> > > The whole section here
> > > > When I attempted to set the rule my router warned me with the following: "Warning: The change you are attempting to make may cause AT&T U-verse TV to stop working properly." I don't use this service, but it appears to me that it could interrupt a lot more communications than that. I have lots of clients in my house that could use http/s protocols, so it seems weird to forward all of them to just this one device.
> > > 
> > > 
> > > Seems to suggest that you have little idea what a port forward is as it will not affect a single client on the same network.
> > 
> > 
> > You may be right, and I might be screwing up the terminology. What I'm talking about is external https requests coming in over external port 443 would be forced to the saltbox. If I have a different https request for a different server on my local network wouldn't that isolate that other server?
> 
> This is a you problem as that is outside of the scope of our project. If you want to do advanced network routing then you would need to know how to handle the proxying required.

Sorry, I wasn't trying to turn this into a support thread. I'm just trying to make sure that some of these restrictions or limiations are understood. So is that that a limitation of saltbox? That if you have other servers using external http or https requests will not be available once the required port forwarding is turned on?

**Comment by saltydk** [comment]
It is a networking thing.

**Comment by dtroose** [comment]
> In the case of ddns and ddclient they are both docker containers, so you would just remove them with the generic
> 
> ```
> docker rm -f containername
> ```

Awesome. And that makes sense when I dug deeper into some of the docker configuration and saw ddclient wasn't listed. Maybe just add that to the page I indicated?

**Comment by dtroose** [comment]
> It is a networking thing.

Agreed. I'll take the support aspects of my install to the discord. Thanks!

**Comment by saltydk** [comment]
> If you use Cloudflare for DNS, use the ddns saltbox role and it will be configured for you. if the ddclient role has been installed, remove it first.

It was telling you to remove it if it was installed, so not sure how else to word that.

**Comment by saltydk** [comment]
Anyways, I am heading off and maybe chaz will reply later as it is still early in the day for him since he is our docs guru.

**Comment by dtroose** [comment]
> > If you use Cloudflare for DNS, use the ddns saltbox role and it will be configured for you. if the ddclient role has been installed, remove it first.
> 
> It was telling you to remove it if it was installed, so not sure how else to word that.

If you use Cloudflare for DNS and the ddclient role has been installed, remove it first with:

`docker rm -f ddclient`

Then use ddns saltbox role and it will be configured for you:

`sb install ddns`
